cogs5e/models/sheet/coinpurse.py

```diff
@@ -0,0 +1,180 @@
+from cogs5e.models.errors import InvalidArgument
+from .mixins import HasIntegrationMixin
+from utils.functions import confirm
+
+CoinTypes = {
+    "pp": {
+        "icon": "<:DDBPlatinum:948681049326624849>",
+        "name": "Platinum",
+        "gSheet": {
+            "v14": "D72",
+            "v2": "D15",
+        }
+    },
+    "gp": {
+        "icon": "<:DDBGold:948681049221775370>",
+        "name": "Gold",
+        "gSheet": {
+            "v14": "D69",
+            "v2": "D12",
+        }
+    },
+    "ep": {
+        "icon": "<:DDBElectrum:948681048932364401>",
+        "name": "Electrum",
+        "gSheet": {
+            "v14": "D66",
+            "v2": "D9",
+        }
+    },
+    "sp": {
+        "icon": "<:DDBSilver:948681049288867930>",
+        "name": "Silver",
+        "gSheet": {
+            "v14": "D63",
+            "v2": "D6",
+        }
+    },
+    "cp": {
+        "icon": "<:DDBCopper:948681049217597480>",
+        "name": "Copper",
+        "gSheet": {
+            "v14": "D60",
+            "v2": "D3",
+        }
+    }
+}
+
+
+class Coinpurse(HasIntegrationMixin):
+    def __init__(self, pp=0, gp=0, ep=0, sp=0, cp=0):
+        super().__init__()
+        self.pp = pp
+        self.gp = gp
+        self.ep = ep
+        self.sp = sp
+        self.cp = cp
+
+    def __str__(self):
+        return "\n".join(self.str_styled(coin_type) for coin_type in CoinTypes)
+
+    def str_styled(self, coin_type):
+        if coin_type not in ("compact", "pp", "gp", "ep", "sp", "cp"):
+            raise ValueError("coin_type must be in ('compact', 'pp', 'gp', 'ep', 'sp', 'cp')")
+
+        if coin_type == 'compact':
+            coin_value = self.total
+            coin_type = 'gp'
+        else:
+            coin_value = getattr(self, coin_type)
+        return f"{CoinTypes[coin_type]['icon']} {coin_type}: {coin_value:,}"
```

Comment:
Rather than `coin_type` including the compact form, we should split this into two methods - `coin_str` and `compact_str`. It's then up to the caller to use the right one based on character options.